### PR TITLE
Fix type hints for token and models.__call__()

### PR DESCRIPTION
**Pull request by leafyoung**
- Added return types for Token's methods. Improving from previous commit, `.name()`/`.symbol()` could return None and causes type checks flagging.
- Remove type hints of `dict` for `models.__call__()` as it could return DTO now.

### CODE REVIEW

File: credmark/cmf/model/context.py [hunk 1]
```diff
@@ -46,7 +46,10 @@ def __init__(self, context, prefix: str, block_number: Union[int, None] = None):
                 self.__doc__ = f'Run a model.\n\nAvailable models: {", ".join(slugs)}'
 
     # run a model. args can be a positional DTO or dict or kwargs
-    def __call__(self, input: Union[DTO, dict, None] = None, return_type=None, **kwargs) -> dict:
+    def __call__(self,
+                 input: Union[DTO, dict, None] = None,
+                 return_type: Union[DTO, dict, None] = None,
+                 **kwargs):
```
Comment (iwyrkore) [inline credmark/cmf/model/context.py:52]: @leafyoung Is it getting the return type right without the hint?

File: credmark/cmf/types/token.py [hunk 2]
```diff
@@ -115,12 +115,16 @@ def try_erc20_property(self, prop_name):
...
 
     @property
-    def symbol(self):
+    def symbol(self) -> str:
         self._load()
         if self._meta.symbol is None:
             symbol_tmp = self.try_erc20_property('symbol')
             if isinstance(symbol_tmp, bytes):
                 symbol_tmp = symbol_tmp.decode('utf-8', errors='strict').replace('\x00', '')
+            elif isinstance(symbol_tmp, str):
```
Comment (iwyrkore) [inline credmark/cmf/types/token.py:124]: Maybe use 

```
elif not isinstance(...):
   raise ModelDataError(...)
```

Same below in name().

Comment (leafyoung) [reply credmark/cmf/types/token.py:125]: good. updated.

File: credmark/cmf/model/context.py (hunk 1 above)
Comment (leafyoung) [reply credmark/cmf/model/context.py]: @iwyrkore Earlier I thought of relying on `self.__context.run_model` without a type hint. Now I have added the type hints.

**Pull request merged**